payments/views/v0/__init__.py

```diff
@@ -385,3 +388,53 @@ def get_queryset(self):
             .order_by("-created_on")
             .all()
         )
+
+
+@extend_schema(
+    description=(
+        "Creates or updates a basket for the current user, "
+        "adding the discount if valid."
+    ),
+    methods=["POST"],
+    request=None,
+    responses=BasketWithProductSerializer,
+)
+@api_view(["POST"])
+@permission_classes((IsAuthenticated,))
+def add_discount_to_basket(request, system_slug: str):
+    """
+    Add a discount to the basket for the currently logged in user.
+
+    Args:
+        system_slug (str): system slug
+
+    POST Args:
+        discount_code (str): discount code to apply to the basket
+
+    Returns:
+        Response: HTTP response
+    """
+    system = IntegratedSystem.objects.get(slug=system_slug)
+    basket = Basket.establish_basket(request, system)
+    discount_code = request.data.get("discount_code")
+
+    try:
+        discount = Discount.objects.get(discount_code=discount_code)
+    except Discount.DoesNotExist:
+        return Response(
+            {"error": "Discount not found"},
+            status=status.HTTP_404_NOT_FOUND,
+        )
+
+    try:
+        basket.apply_discount_to_basket(discount)
+    except ValueError as exc:
+        return Response(
+            {"error": str(exc)},
```

Comment:
CodeQL called this out but I think this is worth fixing also. (The suggestion it provided would be fine with me.)